io/src/csv/read_cells.cpp

```diff
@@ -150,8 +148,13 @@ void read_cells(cell_collection_types::host& cells, std::string_view filename,
 
         // Add the cells to the output.
         for (auto& cell : cellz) {
-            cells.push_back(cell);
-            cells.back().module_link = ddIndex;
+            const std::size_t cellIndex = cells.size();
+            cells.resize(cellIndex + 1);
+            cells.channel0().at(cellIndex) = cell.channel0;
+            cells.channel1().at(cellIndex) = cell.channel1;
+            cells.activation().at(cellIndex) = cell.value;
+            cells.time().at(cellIndex) = cell.timestamp;
+            cells.module_index().at(cellIndex) = ddIndex;
```

Comment:
Strongly consider adding an emplace method to the EDM object; this is too verbose and error-prone.

Reply:
This would be tied to the proxy idea discussed above. Without a proxy type there's nothing better that we can do at the moment. :frowning:

Reply:
Have a look! I think the code is much more visually appealing now. :wink: